physlr/physlr.py

```diff
@@ -400,14 +400,49 @@ def diameter_of_tree(g, weight=None):
         return (u, v, diameter)
 
     @staticmethod
-    def determine_backbones_of_trees(g):
-        "Determine the backbones of the maximum spanning trees."
+    def detect_junctions_of_tree(gcomponent, messages, junction_threshold):
+        """"
+        detect the junctions in the tree,
+        with more than 2 branches larger than junction_threshold.
+        """
+        candidate_junctions = {node
+                               for node in list(gcomponent.nodes)
+                               if gcomponent.degree(node) > 2}
+        junctions = []
+        for candidate in candidate_junctions:
+            candidate_messages = [messages[(candidate, neighbor)]
+                                  for neighbor in gcomponent.neighbors(candidate)]
+            candidate_messages.sort()
+            if candidate_messages[-3] > junction_threshold:
+                junctions.append(candidate)
+        return junctions
+
+    @staticmethod
+    def determine_backbones_of_trees(g, junction_threshold=0):
```

Comment:
```suggestion
    def determine_backbones_of_trees(g, junction_threshold):
```
No default value, and update all callers.